airflow/plugins/operators/scrape_ntd_xlsx.py

```diff
@@ -0,0 +1,174 @@
+    template_fields = ("year", "product", "xlsx_file_url")
+
+    def __init__(
+        self,
+        product: str,
+        xlsx_file_url,
+        year: int,
+        **kwargs,
+    ):
+        self.year = year
+        self.product = product
+        self.xlsx_file_url = xlsx_file_url
+
+        typer.secho(
+            f"file url is {self.xlsx_file_url} and file type is {type(self.xlsx_file_url)}",
+            fg=typer.colors.MAGENTA,
+        )
+
+        # Save initial excel files to the bucket as "raw"
+        self.raw_excel_extract = NtdDataProductXLSXExtractRaw(
+            year=year,
+            product=product + "_raw",
+            file_url=self.xlsx_file_url,
+            filename=f"{year}__{product}_raw.xlsx",
+        )
+
+        super().__init__(**kwargs)
+
+    def execute(self, **kwargs):
+        excel_content = self.raw_excel_extract.fetch_from_ntd_xlsx(
+            self.raw_excel_extract.file_url
+        )
+
+        self.raw_excel_extract.save_content(fs=get_fs(), content=excel_content)
+
+        excel_data = BytesIO(excel_content)
+        df_dict = pd.read_excel(excel_data, sheet_name=None, engine="openpyxl")
+
+        if len(df_dict.keys()) > 1:
+            for key, df in df_dict.items():
+                df = df.rename(make_name_bq_safe, axis="columns")
+                typer.secho(
+                    f"read {df.shape[0]} rows and {df.shape[1]} columns",
+                    fg=typer.colors.MAGENTA,
+                )
+
+                self.clean_gzipped_content = gzip.compress(
+                    df.to_json(orient="records", lines=True).encode()
+                )
+
+                tab_name = ""
+
+                tab_name = make_name_bq_safe(key)
+
+                self.clean_excel_extract = NtdDataProductXLSXExtractClean(
+                    year=self.year,
+                    product=self.product + "/" + tab_name,
+                    filename=f"{self.year}__{self.product}__{tab_name}.jsonl.gz",
+                )
+
+                self.clean_excel_extract.save_content(
+                    fs=get_fs(), content=self.clean_gzipped_content
+                )
+
+        else:
+            df = df.rename(make_name_bq_safe, axis="columns")
+            typer.secho(
+                f"read {df.shape[0]} rows and {df.shape[1]} columns",
+                fg=typer.colors.MAGENTA,
+            )
+
+            self.clean_gzipped_content = gzip.compress(
+                df.to_json(orient="records", lines=True).encode()
+            )
+
+            tab_name = ""
+
+            tab_name = make_name_bq_safe(key)
+
+            self.clean_excel_extract = NtdDataProductXLSXExtractClean(
+                year=self.year,
+                product=self.product + "/" + tab_name,
+                filename=f"{self.year}__{self.product}__{tab_name}.jsonl.gz",
+            )
+
+            self.clean_excel_extract.save_content(
+                fs=get_fs(), content=self.clean_gzipped_content
+            )
```

Comment:
**suggestion:** This feels like it should be broken out to a separate function (e.g. `self.upload_ntd_jsonl(df)` or something) as both this `else` clause and the `for` clause above contain the same code (unless I'm missing some small difference).

```suggestion
        if len(df_dict.keys()) > 1:
            for key, df in df_dict.items():
                self.upload_ntd_jsonl(df)
        else:
            self.upload_ntd_jsonl(df)
    
    def upload_ntd_jsonl(self, df):
        df = df.rename(make_name_bq_safe, axis="columns")
        typer.secho(
            f"read {df.shape[0]} rows and {df.shape[1]} columns",
            fg=typer.colors.MAGENTA,
        )

        self.clean_gzipped_content = gzip.compress(
            df.to_json(orient="records", lines=True).encode()
        )

        tab_name = ""

        tab_name = make_name_bq_safe(key)

        self.clean_excel_extract = NtdDataProductXLSXExtractClean(
            year=self.year,
            product=self.product + "/" + tab_name,
            filename=f"{self.year}__{self.product}__{tab_name}.jsonl.gz",
        )

        self.clean_excel_extract.save_content(
            fs=get_fs(), content=self.clean_gzipped_content
        )
```